test-ciphers.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+console_url=$(oc get console.config.openshift.io cluster --template '{{.status.consoleURL}}')
+if [ -z "${console_url}" ]
+then
+  echo "no console_url, are you connected to a cluster?"
+  exit 1
+fi
+
+# need to format the console_url for s_client
+console_url_without_http=${console_url#"https://"}
+console_url_without_http_with_port="${console_url_without_http}:443"
+SERVER="${console_url_without_http_with_port}"
+
+
+# ensure what we expect works
+# we approve roughly 16 cipher suites
+# but
+valid_cipher_sample=(
+  ECDHE-ECDSA-CHACHA20-POLY1305
+  ECDHE-RSA-AES256-GCM-SHA384
+)
+
+echo "testing valid ciphers...."
+# we want these to successfully connect
+for CIPHER in "${valid_cipher_sample[@]}"
+do
+  # test one, silencing output
+  result=$(echo -n | openssl s_client -connect $SERVER -cipher "${CIPHER}" 2>&1)
+  if [[ "$result" =~ "no cipher match" ]] ; then
```

Comment:
Can we just check the return code instead of looking for a specific message? Presumably it's a zero return code if it works or a non-zero return code in the case of no cipher match

Reply:
So this is actually a little bit tricky.  If I do:

```bash
openssl s_client -connect console-openshift-console.apps.bpetersen.devcluster.openshift.com:443 -cipher ECDHE-ECDSA-CHACHA20-POLY1305
```
I get several layers of output:

```bash 
CONNECTED(00000007)
4678395500:error:14004410:SSL routines:CONNECT_CR_SRVR_HELLO:sslv3 alert handshake failure:/BuildRoot/Library/Caches/com.apple.xbs/Sources/libressl/libressl-22.260.1/libressl-2.6/ssl/ssl_pkt.c:1205:SSL alert number 40
4678395500:error:140040E5:SSL routines:CONNECT_CR_SRVR_HELLO:ssl handshake failure:/BuildRoot/Library/Caches/com.apple.xbs/Sources/libressl/libressl-22.260.1/libressl-2.6/ssl/ssl_pkt.c:585:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 7 bytes and written 0 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID:
    Session-ID-ctx:
    Master-Key:
    Start Time: 1575323630
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
---
```
There are some success and some failures in there, ultimately ending with `status code: 1`.  But, we `CONNECTED`, and I think thats mainly what we are after.  So switching to status code checks always fails.

I can look into another way to write this that perhaps suppresses the other noise if that would be better.

Reply:
Ideally we would use the correct CA so that status code is 0, and then we can just check status code. Presumably the command fails because it's the certificate is not trusted.

Reply:
Can you get the `router-ca` from `openshift-config-managed` and pass it as `-CAFile` to `openssl`?

Reply:
Simpler code issues for openssl seems to be a requested feature. https://github.com/openssl/openssl/issues/5752